src/v2/view-builder/views/webauthn/ChallengeWebauthnView.js

```diff
@@ -18,11 +19,22 @@ const Body = BaseForm.extend({
     const schema = [];
     // Returning custom array so no input fields are displayed for webauthn
     if (webauthn.isNewApiAvailable()) {
+      let shouldUpdateButtonText = false;
+      let title = loc('retry', 'login');
+      if (BrowserFeatures.isSafari()) {
+        title = loc('mfa.challenge.verify', 'login');
+        shouldUpdateButtonText = true;
+      }
       const retryButton = createButton({
         className: 'retry-webauthn button-primary default-custom-button',
-        title: loc('retry', 'login'),
+        title,
         click: () => {
           this.getCredentialsAndSave();
+          // Update button text only when using safari.
+          if (shouldUpdateButtonText) {
```

Comment:
wouldn't it be a little cleaner to invoke `BrowserFeatures.isSafari()` instead of introducing local variable (a.k.a manage state)?

Reply:
I agree.. Was this done so that the text doesn't change on every retry?

Reply:
@haishengwu-okta  I can do that but if I check for Safari then I would be checking again and again. Having said that I feel a user wouldnt cancel that many times normally :) Will update.